packages/components/pie-select/src/index.ts

```diff
@@ -106,10 +131,26 @@ export class PieSelect extends FormControlMixin(RtlMixin(PieElement)) implements
      * Resets the value to the default select value.
      */
     public formResetCallback (): void {
-        const selected = this._select.querySelector('option[selected]');
-        this._select.value = selected?.getAttribute('value') ?? '';
-        this._select.selectedIndex = selected ? this._select.selectedIndex : 0;
-        this._internals.setFormValue(this._select.value);
+        // Flatten a possibly nested options list into a flat one
+        const flatOptions: SelectOptionProps[] = this.options.reduce<SelectOptionProps[]>((acc, option) => {
+            if (option.tag === 'optgroup') {
+                return acc.concat(option.options);
+            }
+            acc.push(option);
+            return acc;
+        }, []);
+
+        // Infer the value to reset to
+        const firstValue = flatOptions ? flatOptions[0].value : '';
```

Comment:
I think an empty array is still truthy, so maybe something like this might be safer?

```suggestion
        const firstValue = flatOptions.length > 0 ? flatOptions[0].value : '';
```

Reply:
Related question, do we have a test case for a completely empty options array? I'm curious to know if it throws an error